pkg/sql/plan2/function/builtin/multi/substr.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/substring"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func Substring(inputVecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
```

Comment:
I think
maybe 2 overloads for Substring is better ?
(
overload 1 : substring(string, int)
overload 2: substring(string, int, int)
)
it's doesn't better

Reply:
OK, I will consider this overload method based on the number of parameters。